internal/cluster/common.go

```diff
@@ -68,7 +75,7 @@ func UpdateCluster(log logrus.FieldLogger, db *gorm.DB, clusterId strfmt.UUID, s
 	// Status is required as well to avoid races between different components.
 	dbReply := db.Model(&common.Cluster{}).Where("id = ? and status = ?", clusterId, srcStatus).Updates(updates)
 
-	if dbReply.Error != nil || dbReply.RowsAffected == 0 {
+	if dbReply.Error != nil || (dbReply.RowsAffected == 0 && !clusterExistsInDB(db, clusterId, updates)) {
```

Comment:
Isn't it the same as ignoring `(dbReply.RowsAffected == 0 && !clusterExistsInDB(db, clusterId, updates))` ? 
We are trying to update cluster so in 99.9% it exists in the DB.

Reply:
If the cluster with the same fields that we want to update is already in the database, it is not an error.  If there is no change in the database record, then gorm will return 0 rows affected, but it is not an error.

Reply:
The same already exists for hosts

Reply:
You compare only id so you don't really know if it's the same or not.

Reply:
No. The updates map is also sent to the function.  So it is part if the "where".